metrics/cmd/heapster-apiserver/app/server.go

```diff
@@ -60,12 +61,11 @@ func NewHeapsterApiServer(s *options.HeapsterRunOptions, metricSink *metricsink.
 }
 
 func newAPIServer(s *options.HeapsterRunOptions) (*genericapiserver.GenericAPIServer, error) {
-	if err := s.SecureServing.MaybeDefaultWithSelfSignedCerts("heapster.kube-system"); err != nil {
+	if err := s.SecureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1")}); err != nil {
```

Comment:
this should not change to "localhost" for the self-signed certs.  IIRC, the newer code has an option for this.  We can fix it in a follow-up, though.